### PR TITLE
Separate headers and response body using CURLINFO_HEADER_SIZE flag

### PR DESCRIPTION
**Pull request by innamhunzai**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | separate headers and response body using CURLINFO_HEADER_SIZE
| Type?             | bug fix
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #[95](https://github.com/PrestaShop/PrestaShop-webservice-lib/issues/95)
| Sponsor company   | NxtLvl B.V.
| How to test?      | N/A

### CODE REVIEW

File: PSWebServiceLibrary.php [hunk 1]
```diff
@@ -192,13 +192,13 @@ protected function executeRequest($url, $curl_params = array())
...
-        if ($index === false && $curl_params[CURLOPT_CUSTOMREQUEST] != 'HEAD') {
+        $headerSize = curl_getinfo($session, CURLINFO_HEADER_SIZE);
+        if ($headerSize === false && $curl_params[CURLOPT_CUSTOMREQUEST] != 'HEAD') {
             throw new PrestaShopWebserviceException('Bad HTTP response ' . $response . curl_error($session));
         }
 
-        $header = substr($response, 0, $index);
-        $body = substr($response, $index + 4);
+        $header = substr($response, 0, $headerSize);
+        $body = substr($response, $headerSize);
```
Comment (matthieu-rolland) [inline PSWebServiceLibrary.php:201]: did you remove the `+ 4` intentionally here ?

**Review by matthieu-rolland — COMMENTED** [review]
just one question

File: PSWebServiceLibrary.php (hunk 1 above)
Comment (innamhunzai) [reply PSWebServiceLibrary.php:201]: yes, it was for skipping 4 these escape chars "\r\n\r\n", we dont need it any more.

**Review by tleon — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @innamhunzai  

Thank for your PR,
This PR is too technical for me, ping @PrestaShop/tech-council, can someone please test this PR ? 

Thanks

**Review by matks — APPROVED** [review]
Tested and validated 👍

**Comment by matks** [comment]
Thank you @innamhunzai and sorry for late merge

**Pull request merged**